complex/src/lib.rs

```diff
@@ -518,10 +518,50 @@ impl<T: Clone + Num> Div<Complex<T>> for Complex<T> {
     }
 }
 
+forward_all_binop!(impl Rem, rem, PartialOrd);
+
+// Attempts to identify the gaussian integer whose product with `modulus`
+// is closest to `self`.
+impl<T: Clone + Num + PartialOrd> Rem<Complex<T>> for Complex<T> {
+    type Output = Complex<T>;
+
+    #[inline]
+    fn rem(self, modulus: Complex<T>) -> Self {
+        let Complex { re, im } = self.clone() / modulus.clone();
+        // This is the gaussian integer corresponding to the true ratio
+        // rounded towards zero.
+        let (re0, im0) = (re.clone() - re % T::one(), im.clone() - im % T::one());
+
+        let zero = T::zero();
+        let one  = T::one();
+        let neg  = zero.clone() - one.clone();
+        // Traverse the 3x3 square of gaussian integers surrounding our
+        // current approximation, and select the one whose product with
+        // `modulus` is closest to `self`.
```

Comment:
Yes, that is a funny example.  Do you have one that would fail lexicographical ordering?

I guess we see why ordering and remainders aren't usually implemented for complex numbers.  But if we do implement them, I think it's a fundamental property that operations on `Complex` values with zero imaginary should match the equivalent operation on the real part alone.